### ISSUE TITLE
duplicate reference target from cim_obj.py  

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#761]
Here is the documentation issue we were talking about a this morning:
```
/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_obj.py:docstring of pywbem.CIMInstanceName:: WARNING: more than one target found for cross-reference u'host': pywbem.WBEMListener.host, pywbem.WBEMConnection.host
/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_obj.py:docstring of pywbem.CIMClassName:: WARNING: more than one target found for cross-reference u'host': pywbem.WBEMListener.host, pywbem.WBEMConnection.host
```

**Comment by andy-maier** [comment]
Both occurrences are defined as "Attributes:", e.g.:
```
class CIMClassName(_CIMComparisonMixin):
    """
    A CIM class path.

    Attributes:

      host (:term:`unicode string`):
        Host and optionally port of the WBEM server containing the CIM
        namespace of the referenced class, in the format:
```

For unknown reasons, Sphinx represents attributes with a link to some target it tries to locate automatically. Because "host" is defined more than once (as properties on `WBEMConnection` and on `WBEMListener`), it displays the warning we see.

So the scope of this issue is larger than the warnings that are shown, it affects all uses of an "Attributes:" section. We have 14 uses of "Attributes:" in the pywbem docstrings.

I see two options for fixing this:
1. Remove the use of "Attributes:" with a statement that the init parameters are shown as attributes. This needs verification, it may not always be entirely true, and we would lose specifics like that string properties can be unicode or byte string as an init parameter, but will always be unicode as the corresponding attribute.
2. Understand why Sphinx tries to create a link for an entity that is defined there, and how that can be avoided.

**Comment by andy-maier** [comment]
I have created issue sphinx-doc/sphinx#4074 to clarify this.

**Comment by andy-maier** [comment]
A solution for us that does not depend on a Sphinx fix for the reported issue would be to convert the documentation of attributes to python properties (with setters, for writable attributes). in addition to not running into the Sphinx issue, this provides better control about input arguments inthe setters, and allows changing input strings into unicode (which is what the descriptions promise anyway, but is not done, see issue #746).

**Comment by andy-maier** [comment]
Fixed in PR #784.
